### PR TITLE
TestFlight only release and promotion

### PR DESCRIPTION
**Pull request by kshann**
Task/Issue URL: https://app.asana.com/0/0/1208234424227124/f
Tech Design URL:
CC:  @ayoy 

**Description**:
- Changes nightly builds to release to TestFlight only
- Adds promote TestFlight to appstore (mirroring MacOS) which gets the latest test flight build and uploads the meta data for that version. Currently only tested with Alpha. 

**Steps to test this PR**:
-  [Initial test flow](https://github.com/duckduckgo/iOS/actions/runs/11491500989/job/31983979375) (without upload meta data).
   - This iteration of the flow only checked the version and build number to upload without doing the upload
   - For Alpha it found `Latest upload for version 7.142.0 on ios platform is build: 1`
   - For Stable it found `Latest upload for version 7.142.0 on ios platform is build: 0`
- The nightly job ran (conveniently) to update the alpha build in TestFlight, and a rerun of the [promote flow](https://github.com/duckduckgo/iOS/actions/runs/11491747964/job/31984649252)  showed it found updated build number of 2 which we expect
- [Final run](https://github.com/duckduckgo/iOS/actions/runs/11492279658) failed due to Alpha not quite setup to accept submissions due to metadata conflicts with stable, but it did successfully update the version and some of the metadata, so confident this will work for stable builds.

Before 
image

After promotion run
image

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -168,6 +168,32 @@ lane :alpha_adhoc do |options|
...
 end
 
+
+
+desc 'Promotes the latest TestFlight build to App Store without submitting for review'
+lane :promote_latest_testflight_to_appstore do |options|
+
+  app_identifier = options[:alpha] ? "com.duckduckgo.mobile.ios.alpha" : "com.duckduckgo.mobile.ios"
+
+  latest_build_info = latest_testflight_build_number(
```
Comment (ayoy) [inline fastlane/Fastfile]: latest_build_info is unused and can be removed:
```suggestion
  latest_testflight_build_number(
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -36,10 +36,10 @@ jobs:
       id: destination
       run: |
         INPUT_DESTINATION=${{ github.event.inputs.destination }}
-        echo "destination=${INPUT_DESTINATION:-"appstore"}" >> $GITHUB_OUTPUT
+        echo "destination=${INPUT_DESTINATION:-"testflight"}" >> $GITHUB_OUTPUT
 
     - name: Assert release branch
-      if: steps.destination.outputs.destination == 'appstore'
+      if: steps.destination.outputs.destination == 'appstore' || steps.destination.outputs.destination == 'testflight'
```
Comment (ayoy) [inline .github/workflows/release.yml]: `appstore` and `testflight` are the only options on iOS, so we can remove the `if:` statement altogether 👍

**Review by ayoy — APPROVED** [review]
LGTM! Just 2 minor comments. Great job 💪

**Comment by kshann** [comment]
Will merge Monday after this weeks release.

**Review by ankushkushwaha — APPROVED** [review]

**Pull request merged**